server/src/main/java/com/kritsit/casetracker/server/datalayer/UserRepository.java

```diff
@@ -0,0 +1,62 @@
+package com.kritsit.casetracker.server.datalayer;
+
+import java.util.Map;
+
+import com.kritsit.casetracker.server.domain.model.Permission;
+import com.kritsit.casetracker.server.domain.model.Staff;
+
+public class UserRepository implements IUserRepository {
+	
+	private final IPersistenceService db;
+	
+	public UserRepository(){
+		//Just until we have mockin \ DI
+		this.db = new DatabasePersistence();
+	}
+	
+	public UserRepository(IPersistenceService db){
+		this.db = db;
+	}
+	
+    public long getPasswordSaltedHash(String username) throws RowToModelParseException {
+    	try {
+	        String sql = "SELECT passwordHash FROM staff WHERE username=\'" 
+	            + username + "\';";
+	        Map<String, String> rs = db.executeQuery(sql);
+	        return Long.parseLong(rs.get("passwordHash"));
```

Comment:
I know what you mean, but I'm more scared of "special values" returned by a method that another method (or a series of methods that call the first) needs to interpret. My idea was to throw exceptions that are catched in a sort of event bus. Think of a huge try \ catch on the border of the server (except we'd use an aspect to manage it to make the code cleaner).
But if you want we can return a negative value, that's easy enough to understand.